In [1]:
!pip install scikit-learn==1.2.2
!pip install pandas
!pip install pyecg --no-deps
!pip install pyprep scipy wandb
!pip install ishneholterlib

ERROR: Could not find a version that satisfies the requirement 1.2.2 (from versions: none)

[notice] A new release of pip is available: 25.0.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip
ERROR: No matching distribution found for 1.2.2



[notice] A new release of pip is available: 25.0.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


  Using cached pyECG-0.0.9.1-py2.py3-none-any.whl.metadata (3.1 kB)
Using cached pyECG-0.0.9.1-py2.py3-none-any.whl (14 kB)



[notice] A new release of pip is available: 25.0.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


  Using cached pyprep-0.5.0-py3-none-any.whl.metadata (7.3 kB)
  Using cached wandb-0.21.3-py3-none-win_amd64.whl.metadata (10 kB)
  Using cached click-8.2.1-py3-none-any.whl.metadata (2.5 kB)
  Using cached gitpython-3.1.45-py3-none-any.whl.metadata (13 kB)
  Using cached protobuf-6.32.0-cp310-abi3-win_amd64.whl.metadata (593 bytes)
  Using cached pydantic-2.11.7-py3-none-any.whl.metadata (67 kB)
  Using cached sentry_sdk-2.35.2-py2.py3-none-any.whl.metadata (10 kB)
  Using cached typing_extensions-4.15.0-py3-none-any.whl.metadata (3.3 kB)
  Using cached gitdb-4.0.12-py3-none-any.whl.metadata (1.2 kB)
  Using cached annotated_types-0.7.0-py3-none-any.whl.metadata (15 kB)
  Using cached typing_inspection-0.4.1-py3-none-any.whl.metadata (2.6 kB)
  Using cached smmap-5.0.2-py3-none-any.whl.metadata (4.3 kB)
Using cached pyprep-0.5.0-py3-none-any.whl (34 kB)
Using cached wandb-0.21.3-py3-none-win_amd64.whl (18.7 MB)
Using cached click-8.2.1-py3-none-any.whl (102 kB)
Using cached gitpython


[notice] A new release of pip is available: 25.0.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


  Using cached ishneholterlib-2020.5.29-py3-none-any.whl.metadata (356 bytes)
  Using cached crccheck-1.3.1-py3-none-any.whl.metadata (5.5 kB)
Using cached ishneholterlib-2020.5.29-py3-none-any.whl (8.4 kB)
Using cached crccheck-1.3.1-py3-none-any.whl (24 kB)


ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pyecg 0.0.9.1 requires pytest, which is not installed.
pyecg 0.0.9.1 requires pytest-cov, which is not installed.
pyecg 0.0.9.1 requires wfdb==2.2.1, which is not installed.
pyecg 0.0.9.1 requires ishneholterlib==2017.4.11, but you have ishneholterlib 2020.5.29 which is incompatible.

[notice] A new release of pip is available: 25.0.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [1]:
import pyprep
import scipy
import wandb
import pyecg

print("PyPREP:", pyprep.__version__)
print("SciPy:", scipy.__version__)
print("wandb:", wandb.__version__)
print("pyECG:", pyecg.__version__)

c:\Users\cakra\AppData\Local\Programs\Python\Python313\Lib\site-packages\pyecg\__init__.py:2: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  from pkg_resources import get_distribution, DistributionNotFound


PyPREP: 0.5.0
SciPy: 1.15.2
wandb: 0.21.3
pyECG: 0.0.9.1


### PyPREP
Library untuk preprocessing sinyal EEG. Digunakan untuk membersihkan sinyal EEG dari noise, artefak, dan baseline wander menggunakan pipeline PREP, sehingga menghasilkan data yang lebih akurat untuk analisis neuroscience atau machine learning.

### SciPy
Library Python untuk komputasi ilmiah. Menyediakan fungsi untuk perhitungan matematis, statistik, pemrosesan sinyal (misalnya FFT, filtering), optimisasi, dan integrasi. Cocok untuk analisis data teknis dan ilmiah.

### wandb (Weights & Biases)
Platform untuk pelacakan eksperimen machine learning. Memungkinkan pengguna mencatat metrik (loss, akurasi), hyperparameter, dan visualisasi performa model secara real-time di dashboard online.

### pyECG
Library untuk analisis sinyal ECG. Mendukung pembacaan data ECG, preprocessing, ekstraksi fitur (seperti deteksi QRS), dan analisis sinyal jantung untuk aplikasi kesehatan atau penelitian biometrik.

# Bagian 2: Etika, Hukum, dan Dampak Lingkungan AI


## 1. Berikan contoh tentang tidakan melanggar etika dan hukum tentang penggunaan kecerdasan buatan. Sertakan referensi yang digunakan.

Jawab: seorang pengacara di Victoria, Australia, menjadi yang pertama dalam mendapatkan sanksi profesional karena menggunakan AI untuk menghasilkan kutipan hukum palsu dan tidak dapat melakukan verifikasi pada sidang. sehingga sertifikat praktik diubah dan tidak bisa membuka praktik atau kelola dana klien serta harus bekerja dibawah pengawasan selama 2 tahun. Sehingga regulasi menggunakan verifikasi manual dan menolak larangan secara total penggunaan AI.

sumber referensi:
https://www.theguardian.com/law/2025/sep/03/lawyer-caught-using-ai-generated-false-citations-in-court-case-penalised-in-australian-first

## 2. Berikan contoh tentang dampak energi dan lingkungan terhadap pemanfaatan kecerdasan buatan dan bagaimana cara mengatasinya (Anda dapat memberikan pendapat pribadi). Sertakan referensi yang digunakan.

Jawab:

a. Konsumsi Daya yang sangat tinggi
   Data center yang digunakan untuk menjalankan AI akan membutuhkan daya dengan skala besar. Bahkan, badan International Energy Agency (IEA) memperkirakan konsumsi listrik global dari pusat data AI akan lebih dari dua kali lipat hingga 2030, dan bisa melampaui sektor berat seperti baja dan semen
   
   Sumber referensi:
   https://www.theguardian.com/technology/2025/apr/10/energy-demands-from-ai-datacentres-to-quadruple-by-2030-says-report

b. Pertumbuhan AI yang semakin pesat akan membebani jaringan listrik
   dan sistem pendinginan. Bahkan wilayah seperti Northern Virginia (AS) dan Irlandia sudah mengalami keterbatasan pasokan energi sehingga memperlambat ekspansi yang ada pada data center

   Sumber referensi: https://www.techradar.com/pro/ais-infrastructure-problem-is-bigger-than-we-think-heres-how-to-solve-it?

c. Operasional dan produksi yang terus bertumbuh akan menghasilkan emisi karbon dan emisi udara. Dengan contoh penggunaan model AI seperti Llama 3.1 dapat menghasilkan polusi yang sama dengan 10.000 perjalanan mobil di Los Angeles–New York. Yang mengakibatkan biaya kesehatan pada publik di AS akan berpotensi mencapai US$20 miliar per tahun pada 2030.

Sumber referensi: https://arxiv.org/abs/2412.06288

Dengan cara mengatasi:

a. Setiap perusahaan AI harus memberikan pelaporan mengenai emisi karbon, konsumsi air dan penggunaan air dari pembangunan AI, dengan menggunakan standar ISO

Sumber referensi: https://www.weforum.org/stories/2025/06/how-ai-use-impacts-the-environment/

https://www.theguardian.com/technology/2025/apr/10/energy-demands-from-ai-datacentres-to-quadruple-by-2030-says-report

b. Menggunakan energi yang terbarukan dengan membuat pengalihan data center ke sumber energi yang bersih dengan mengalihkan ke panel surya, nuklir, bahkan diarahkan ke startup energi yang terbarukan.

Sumber referensi: https://time.com/7294803/ai-revolution-energy-revolution/

